src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,686 @@
+        IERC20(asset).transferFrom(from, to, amount);
+    }
+
+    function _swapExactTokensForTokens(IERC20 inToken, IERC20 outToken, uint256 amountIn, address to)
+        internal
+        virtual
+        returns (uint256 amountOut)
+    {
+        uint256 price;
+        if (inToken == token0) {
+            require(outToken == token1, "ARM: Invalid out token");
+            price = traderate0;
+        } else if (inToken == token1) {
+            require(outToken == token0, "ARM: Invalid out token");
+            price = traderate1;
+        } else {
+            revert("ARM: Invalid in token");
+        }
+        amountOut = amountIn * price / PRICE_SCALE;
+
+        // Transfer the input tokens from the caller to this ARM contract
+        _transferAssetFrom(address(inToken), msg.sender, address(this), amountIn);
+
+        // Transfer the output tokens to the recipient
+        _transferAsset(address(outToken), to, amountOut);
+    }
+
+    function _swapTokensForExactTokens(IERC20 inToken, IERC20 outToken, uint256 amountOut, address to)
+        internal
+        virtual
+        returns (uint256 amountIn)
+    {
+        uint256 price;
+        if (inToken == token0) {
+            require(outToken == token1, "ARM: Invalid out token");
+            price = traderate0;
+        } else if (inToken == token1) {
+            require(outToken == token0, "ARM: Invalid out token");
+            price = traderate1;
+        } else {
+            revert("ARM: Invalid in token");
+        }
+        amountIn = ((amountOut * PRICE_SCALE) / price) + 1; // +1 to always round in our favor
+
+        // Transfer the input tokens from the caller to this ARM contract
+        _transferAssetFrom(address(inToken), msg.sender, address(this), amountIn);
+
+        // Transfer the output tokens to the recipient
+        _transferAsset(address(outToken), to, amountOut);
+    }
+
+    /**
+     * @notice Set exchange rates from an operator account from the ARM's perspective.
+     * If token 0 is WETH and token 1 is stETH, then both prices will be set using the stETH/WETH price.
+     * @param buyT1 The price the ARM buys Token 1 from the Trader, denominated in Token 0, scaled to 36 decimals.
+     * From the Trader's perspective, this is the sell price.
+     * @param sellT1 The price the ARM sells Token 1 to the Trader, denominated in Token 0, scaled to 36 decimals.
+     * From the Trader's perspective, this is the buy price.
+     */
+    function setPrices(uint256 buyT1, uint256 sellT1) external onlyOperatorOrOwner {
+        // Limit funds and loss when called by the Operator
+        if (msg.sender == operator) {
+            require(sellT1 >= PRICE_SCALE - MAX_PRICE_DEVIATION, "ARM: sell price too low");
+            require(buyT1 <= PRICE_SCALE + MAX_PRICE_DEVIATION, "ARM: buy price too high");
+        }
+        uint256 _traderate0 = 1e72 / sellT1; // base (t0) -> token (t1)
```

Comment:
Do having these temporary variables get us any clarity? They are each only used once.

Reply:
We could change this function's parmeter names to `baseToTokenRate` and `tokenToBaseRate`

Reply:
The prices take a while to understand so any change that can help others is good. I've changed to

```Solidity
function setPrices(uint256 buyT1, uint256 sellT1) external onlyOperatorOrOwner {
    // Limit funds and loss when called by the Operator
    if (msg.sender == operator) {
        require(sellT1 >= PRICE_SCALE - MAX_PRICE_DEVIATION, "ARM: sell price too low");
        require(buyT1 <= PRICE_SCALE + MAX_PRICE_DEVIATION, "ARM: buy price too high");
    }
    _setTraderates(
        PRICE_SCALE * PRICE_SCALE / sellT1, // base (t0) -> token (t1)
        buyT1 // token (t1) -> base (t0)
    );
}

function _setTraderates(uint256 _baseToTokenRate, uint256 _tokenToBaseRate) internal {
    require((PRICE_SCALE * PRICE_SCALE / (_baseToTokenRate)) > _tokenToBaseRate, "ARM: Price cross");
    traderate0 = _baseToTokenRate;
    traderate1 = _tokenToBaseRate;

    emit TraderateChanged(_baseToTokenRate, _tokenToBaseRate);
}
```